### PR TITLE
Use NoThunks to verify no memory leaks in PeerMetrics

### PR DESCRIPTION
**Pull request by coot**
- ouroboros-network-framework: added NoThunk orphaned instances
- peer-metrics: use NoThunks in PeerMetrics

### CODE REVIEW

**Review by bolt12 — APPROVED** [review]

**Comment by coot** [comment]
Some test were failing because we also need to evaluate arguments of `newPeerMetrics'` to normal form.  This fixes it adding a bunch of `NFData` & `NoThunk` instances.

We should add `NFData` instance to `IP` in the `iproute` package (and `NoThunk` instance too).

File: cabal.project [hunk 1]
```diff
@@ -42,7 +42,7 @@ package io-classes
   flags: +asserts
 
 package strict-stm
-  flags: +asserts
+  flags: +asserts +checktvarinvariant
```
Comment (bolt12) [inline cabal.project:45]: should this be 

```suggestion
  flags: +asserts +checktvarinvariants
```

File: nix/ouroboros-network.nix [hunk 2]
```diff
@@ -50,8 +50,8 @@ let
           });
         }
         {
-          packages.strict-stm.components.library.configureFlags = lib.mkForce
-            (if checkTVarInvariant then [ "-f checktvarinvariant" ] else [ ]);
+          packages.strict-checked-vars.components.library.configureFlags = lib.mkForce
+            (if checkTVarInvariants then [ "-f+checktvarinvariants" ] else [ ]);
```
Comment (bolt12) [inline nix/ouroboros-network.nix:54]:
```suggestion
            (if checkTVarInvariants then [ "-f +checktvarinvariants" ] else [ ]);
```

File: cabal.project (hunk 1 above)
Comment (coot) [reply cabal.project:45]: In that commit we're still using `strict-stm` which used `checktvarinvariant`; unfortunatelly the `strict-checkd-vars` package renamed the flag to `checktvarinvariants`.

**Pull request merged**